### ISSUE TITLE
solveset(Abs(x) + 1, x) for positive `x` returns `set([-1])`

### ISSUE THREAD

**Issue by gxyd** [sympy/sympy#9598]
```
>>> x = Symbol('x', positive=True)
>>> from sympy.solvers.solveset import solveset
>>> solveset(Abs(x) + 1, x)
set([-1])
```

I guess it should returns `set([])`
The problem seems to be `Abs(x)` is automatically converted to `x` and later on it is not taken care to remove the solutions which are `non-positive`.

**Comment by aktech** [comment]
Possible duplicate of #9588

**Comment by aktech** [comment]
BTW, It returns FiniteSet(1), i.e. Sympy set, not Python Set.

**Comment by aktech** [comment]
Fixed via #9628 

```python
In [9]: solveset(Abs(x) + 1, x)
Out[9]: EmptySet()
```
